xrpl/wallet/wallet_generation.py

```diff
@@ -48,33 +59,41 @@ def generate_faucet_wallet(client: Client, debug: bool = False) -> Wallet:
         starting_balance = 0
 
     # Ask the faucet to send funds to the given address
-    post(url=FAUCET_URL, json={"destination": address})
+    post(url=faucet_url, json={"destination": address})
```

Comment:
This call can return an error status code like 400 or 503 without raising. We should catch the result and check those.